retriever/__init__.py

```diff
@@ -4,224 +4,9 @@
-            file[0] != "_" and (
-                (not isfile(join(search_path, file[:-5] + '.py'))) or (
-                    isfile(join(search_path, file[:-5] + '.py')) and (
-                        getmtime(join(search_path, file[:-5] + '.py')) < getmtime(
-                            join(search_path, file)))) or force_compile)]
-        for script in to_compile:
-            script_name = '.'.join(script.split('.')[:-1])
-            compile_json(join(search_path, script_name))
-
-        files = [file for file in os.listdir(search_path)
-                 if file[-3:] == ".py" and file[0] != "_" and
-                 '#retriever' in ' '.join(open(join(search_path, file), 'r').readlines()[:2]).lower()]
-
-        for script in files:
-            script_name = '.'.join(script.split('.')[:-1])
-            if script_name not in loaded_scripts:
-                loaded_scripts.append(script_name)
-                file, pathname, desc = imp.find_module(script_name, [search_path])
-                try:
-                    new_module = imp.load_module(script_name, file, pathname, desc)
-                    if hasattr(new_module.SCRIPT, "retriever_minimum_version"):
-                        # a script with retriever_minimum_version should be loaded
-                        # only if its compliant with the version of the retriever
-                        if not parse_version(VERSION) >=  parse_version("{}".format(
-                                new_module.SCRIPT.retriever_minimum_version)):
-                            print("{} is supported by Retriever version {}".format(script_name,
-                                                                                   new_module.SCRIPT.retriever_minimum_version))
-                            print("Current version is {}".format(VERSION))
-                            continue
-                    # if the script wasn't found in an early search path
-                    # make sure it works and then add it
-                    new_module.SCRIPT.download
-                    modules.append(new_module)
-                except Exception as e:
-                    sys.stderr.write("Failed to load script: %s (%s)\nException: %s \n" % (
-                        script_name, search_path, str(e)))
-    return modules
-
-
-def SCRIPT_LIST(force_compile=False):
-    return [module.SCRIPT for module in MODULE_LIST(force_compile)]
-
-
-def ENGINE_LIST():
-    from retriever.engines import engine_list
-    return engine_list
-
-
-def set_proxy():
-    """Check for proxies and makes them available to urllib"""
-    proxies = ["https_proxy", "http_proxy", "ftp_proxy",
-               "HTTP_PROXY", "HTTPS_PROXY", "FTP_PROXY"]
-    for proxy in proxies:
-        if os.getenv(proxy):
-            if len(os.environ[proxy]) != 0:
-                for i in proxies:
-                    os.environ[i] = os.environ[proxy]
-                break
 
+create_dir()
```

Comment:
We should rename this to something like `create_retriever_home` or `create_home_dir`